src/meta/src/rpc/ddl_controller.rs

```diff
@@ -764,6 +765,16 @@ impl DdlController {
         stream_job: &StreamingJob,
         internal_tables: Vec<Table>,
     ) -> MetaResult<()> {
+        let info = serde_json::json!({"id": stream_job.id(), "name": stream_job.name(), "definition": stream_job.definition()}).to_string();
+        let event_log = new_event_log(
+            risingwave_pb::meta::event_log::EventType::CreateStreamJobFail,
+            info,
+        );
+        let _ = self
+            .env
+            .event_log_manager_ref()
+            .add_event_logs(vec![event_log]);
```

Comment:
Not all branch will call this, for instance if cluster full cluster restart, and stream job is cleaned by `clean_dirty_tables` and `clean_dirty_fragments`, it will not call this function.

This will catch the case where failure is caused by CN exit, or perhaps network issues. But if meta node crash it will not.

We can also trace `clean_dirty_tables` / `clean_dirty_fragments` on recovery, if we wish to log stream jobs which were cleaned due to meta node crash.

Reply:
> We can also trace clean_dirty_tables / clean_dirty_fragments on recovery, if we wish to log stream jobs which were cleaned due to meta node crash.

Will do. IIUC one problem with clean_dirty_tables is it traces table/mv being created, but not sink/index, according to `list_persisted_creating_tables`.

Reply:
Fixed. For a failed creation, either log it in `cancel_stream_job` (for index/sink/table_with_source) or `clean_dirty_tables` (for table_without_source/mv), but not both with best efforts.

Reply:
I made another change: `cancel_stream_job` will create an event log of CREATE_STREAM_JOB_FAIL. `clean_dirty_tables` will create an event log of `DIRTY_STREAM_JOB_CLEAR`.

That means if a table or mv creation failed, there would be 2 event log added. My motivation is to let `cancel_stream_job` provide concrete error, which `clean_dirty_tables` has no idea of.